python/pyhie/allspark/model/glm5.py

```
@@ -0,0 +1,1121 @@
+            # Optional DSA indexer: computes top-k token indices for sparse attention
+            dsa_indexer = None
+            if self.has_dsa_indexer:
+                dsa_attrs = {
+                    "index_head_dim": self.index_head_dim,
+                    "index_n_heads": self.index_n_heads,
+                    "index_topk": self.index_topk,
+                    "q_lora_rank": q_lora_rank,
+                    "rope_base": self.rope_base,
+                    "rope_interleave": 1 if rope_interleave else 0,
+                }
+                dsa_indexer = DSAIndexer(
```

Flake8 (7.3.0)

[error] 'DSAIndexer' may be undefined, or defined from star imports: ..quantization, .model_base, .quantization_utils

(F405)